batch-models/src/main/scala/org/ekstep/analytics/dashboard/report/warehouse/DataWarehouseModel.scala

```diff
@@ -48,28 +48,140 @@ object DataWarehouseModel extends IBatchModelTemplate[String, DummyInput, DummyO
     val dwPostgresUrl = s"jdbc:postgresql://${conf.dwPostgresHost}/${conf.dwPostgresSchema}"
     val appPostgresUrl = s"jdbc:postgresql://${conf.appPostgresHost}/${conf.appPostgresSchema}"
 
-
     val user_details = spark.read.option("header", "true")
       .csv(s"/tmp/${conf.userReportPath}/${today}-warehouse")
-    saveDataframeToPostgresTable(user_details, dwPostgresUrl, conf.dwUserTable, conf.dwPostgresUsername, conf.dwPostgresCredential)
 
-    val cbp_details = spark.read.option("header", "true")
-      .csv(s"/tmp/${conf.cbaReportPath}/${today}-warehouse")
-    saveDataframeToPostgresTable(cbp_details, dwPostgresUrl, conf.dwCourseTable, conf.dwPostgresUsername, conf.dwPostgresCredential)
+    postgresConnection(conf.dwUserTable)
+    saveDataframeToPostgresTable_With_Append(user_details, dwPostgresUrl, conf.dwUserTable, conf.dwPostgresUsername, conf.dwPostgresCredential)
+
+
+    var cbp_details = spark.read.option("header", "true")
+      .csv(s"/tmp/${conf.courseReportPath}/${today}-warehouse")
+
+    cbp_details = cbp_details
+      .withColumn("resource_count", col("resource_count").cast("int"))
+      .withColumn("total_certificates_issued", col("total_certificates_issued").cast("int"))
+      .withColumn("cbp_rating", col("cbp_rating").cast("float"))
+      .select(
```

Comment:
why is this select required? all the df columns are necessary and you should just modify the columns that need typecasting like you have done for resource_count, cbp_rating etc.